### DISCUSSION TITLE
recompile with a larger setting for MI_HEAP_REGION_MAX_SIZE

### DISCUSSION THREAD

**Discussion by bowmanjeffs** [ablab/spades#1308] [Q&A]
I'm receiving the error:
`mimalloc: warning: maximum regions used: 256 GiB (perhaps recompile with a larger setting for MI_HEAP_REGION_MAX_SIZE)
`
and am attempting to follow that guidance but am not certain where to change this setting.  I see the variable defined in `ext/src/mimalloc/src/region.c`  so my initial thought is to modify the default value there and recompile.  Curious if anyone has tried this and that's sufficient, or if any other parameters should be changed to avoid the memory allocation error.

**Comment by asl** [comment]
@bowmanjeffs It is not an error and just a warning. The memory allocation errors usually comes from:
 - Well, lack of required memory for the job
 - Misconfigured server

**Comment by bowmanjeffs** [comment]
Thanks!  In case it helps others I did recompile after editing the line referenced above to `define MI_HEAP_REGION_MAX_SIZE    (512 * GiB)` and that stopped the warning.  Sounds like no memory limit was actually reached though.

**Reply by asl** [reply]
The warning is almost always benign.
